pattern/pattern.go

```diff
@@ -20,6 +20,15 @@ import (
 // Not all functions change their behavior with all of the options below.
 type Mode uint
 
+type InvalidGlobError struct {
```

Comment:
How about `pattern.SyntaxError` rather than `pattern.InvalidGlobError`? I don't think we will ever need any kind of "invalid pattern" error that isn't related to syntax, and I would rather avoid the name "glob" as the API is centered around the name "pattern" already - especially in the package name.